lntest/bitcoind.go

```diff
@@ -4,13 +4,15 @@
 package lntest
 
 import (
+	"testing"
+
 	"github.com/btcsuite/btcd/chaincfg"
 )
 
 // NewBackend starts a bitcoind node with the txindex enabled and returns a
 // BitcoindBackendConfig for that node.
-func NewBackend(miner string, netParams *chaincfg.Params) (
-	*BitcoindBackendConfig, func() error, error) {
+func NewBackend(t *testing.T, miner string, netParams *chaincfg.Params) (
```

Comment:
We're currently in the process of completely re-writing the itest framework. Could you please drop this whole commit so we don't incur a lot of rebase conflicts with https://github.com/lightningnetwork/lnd/pull/6759 and its follow-up PRs. Feel free to add comments there to switch over to using `t.Cleanup()`.